invokeai/frontend/web/src/features/controlLayers/hooks/useExportCanvasToPSD.ts

```diff
@@ -0,0 +1,144 @@
+  const canvasManager = useCanvasManagerSafe();
+
+  const exportCanvasToPSD = useCallback(async () => {
+    try {
+      if (!canvasManager) {
+        toast({
+          id: 'CANVAS_MANAGER_NOT_AVAILABLE',
+          title: t('toast.canvasManagerNotAvailable'),
+          status: 'error',
+        });
+        return;
+      }
+
+      // Get visible raster layer adapters using the compositor module
+      const adapters = canvasManager.compositor.getVisibleAdaptersOfType('raster_layer');
+
+      if (adapters.length === 0) {
+        toast({
+          id: 'NO_VISIBLE_RASTER_LAYERS',
+          title: t('toast.noVisibleRasterLayers'),
+          description: t('toast.noVisibleRasterLayersDesc'),
+          status: 'warning',
+        });
+        return;
+      }
+
+      log.debug(`Exporting ${adapters.length} visible raster layers to PSD`);
+
+      // Get the union rect of all adapters using the compositor module
+      const visibleRect = canvasManager.compositor.getRectOfAdapters(adapters);
+
+      // Validate canvas dimensions
+      if (visibleRect.width <= 0 || visibleRect.height <= 0) {
+        toast({
+          id: 'INVALID_CANVAS_DIMENSIONS',
+          title: t('toast.invalidCanvasDimensions'),
+          status: 'error',
+        });
+        return;
+      }
+
+      if (visibleRect.width > 30000 || visibleRect.height > 30000) {
+        toast({
+          id: 'CANVAS_TOO_LARGE',
+          title: t('toast.canvasTooLarge'),
+          description: t('toast.canvasTooLargeDesc'),
+          status: 'error',
+        });
+        return;
+      }
+
+      log.debug(`PSD canvas dimensions: ${visibleRect.width}x${visibleRect.height}`);
+
+      // Create PSD layers from visible raster layer adapters
+      const psdLayers = await Promise.all(
+        adapters.map((adapter, index) => {
+          const layer = adapter.state;
+          const canvas = adapter.getCanvas();
+          const layerPosition = adapter.state.position;
+
+          const layerDataPSD = {
+            name: layer.name || `Layer ${index + 1}`,
+            left: Math.floor(layerPosition.x - visibleRect.x),
+            top: Math.floor(layerPosition.y - visibleRect.y),
+            right: Math.floor(layerPosition.x - visibleRect.x + canvas.width),
+            bottom: Math.floor(layerPosition.y - visibleRect.y + canvas.height),
+            opacity: Math.floor(layer.opacity * 255),
+            hidden: false,
+            blendMode: 'normal' as const,
+            canvas: canvas,
+          };
+
+          log.debug(
+            `Layer "${layerDataPSD.name}": ${layerDataPSD.left},${layerDataPSD.top} to ${layerDataPSD.right},${layerDataPSD.bottom}`
+          );
+
+          return layerDataPSD;
+        })
+      );
+
+      // Create PSD document
+      const psd = {
+        width: visibleRect.width,
+        height: visibleRect.height,
+        channels: 3, // RGB
+        bitsPerChannel: 8,
+        colorMode: 3, // RGB mode
```

Comment:
Use the enum provided by `ag-psd` for `colorMode`.

Reply:
Running into something about ColorMode being a const enum and the TypeScript configuration has isolatedModules enabled. What's proposal?